File: src/test/java/com/parmet/squashlambdas/cal/EventManagerTest.java
package com.parmet.squashlambdas.cal;

import com.google.common.collect.ImmutableSet;
import com.parmet.squashlambdas.activity.Court;
import com.parmet.squashlambdas.activity.Match;
import com.parmet.squashlambdas.testutil.ConfiguredTest;
import java.time.Instant;
import org.junit.Ignore;
import org.junit.Test;

@Ignore // These actually modify a calendar
public class EventManagerTest extends ConfiguredTest {
  @Test
  public void testCreateMatch() throws Exception {
    manager().create(
        new Match(
            Court.COURT_3,
            Instant.parse("2018-06-25T23:30:00Z"),
            Instant.parse("2018-06-26T00:15:00Z"),
            ImmutableSet.of()));
  }

  @Test
  public void testUpdateMatch() throws Exception {
    manager().update(
        new Match(
            Court.COURT_3,
            Instant.parse("2018-06-25T23:30:00Z"),
            Instant.parse("2018-06-26T00:15:00Z"),
            ImmutableSet.of("Logan Ramseyer")));
  }

  @Test
  public void testDeleteMatch() throws Exception {
    manager().delete(
        new Match(
            Court.COURT_3,
            Instant.parse("2018-06-25T23:30:00Z"),
            Instant.parse("2018-06-26T00:15:00Z"),
            ImmutableSet.of("Logan Ramseyer")));
  }

  private EventManager manager() {
    return new EventManager(calendar(), config().getString("google.calendarId"));
  }
}

File: src/test/kotlin/com/parmet/squashlambdas/cal/EventManagerTest.kt
package com.parmet.squashlambdas.cal

import com.parmet.squashlambdas.activity.Court
import com.parmet.squashlambdas.activity.Match
import com.parmet.squashlambdas.activity.Player
import com.parmet.squashlambdas.testutil.ConfiguredTest
import org.junit.Ignore
import org.junit.Test
import java.time.Instant

@Ignore // These actually modify a calendar
class EventManagerTest : ConfiguredTest() {
    @Test
    fun `create creates a match`() {
        manager().create(
            Match(
                Court.Court3,
                Instant.parse("2018-06-25T23:30:00Z"),
                Instant.parse("2018-06-26T00:15:00Z"),
                setOf()
            )
        )
    }

    @Test
    fun `update updates a match`() {
        manager().update(
            Match(
                Court.Court3,
                Instant.parse("2018-06-25T23:30:00Z"),
                Instant.parse("2018-06-26T00:15:00Z"),
                setOf(Player(name = "Logan Ramseyer"))
            )
        )
    }

    @Test
    fun `delete deletes a match`() {
        manager().delete(
            Match(
                Court.Court3,
                Instant.parse("2018-06-25T23:30:00Z"),
                Instant.parse("2018-06-26T00:15:00Z"),
                setOf(Player(name = "Logan Ramseyer"))
            )
        )
    }

    private fun manager() =
        EventManagerImpl(calendar, config.getString("google.cal.calendarId"))
}
